tests/multiple_listening_addresses.rs

```diff
@@ -30,8 +30,22 @@ fn multiple_concurrent_ephemeral_listening_addresses_on_same_ip() {
 
         let (first, second) = futures::future::join(first, second).await;
 
-        // one should succeed and other should fail
-        assert_eq!(first.is_ok(), second.is_err());
+        // before an Swarm alike api on the background task to make sure we attempt to modify the
+        // background task twice before the swarm gets to poll, this will forever produce one or two
+        // successes.
+        //
+        // with two attempts without polling the swarm in the between:
```

Comment:
nit(grammar): "...the swarm is in between:"

Reply:
I literally meant "in the between" as concurrent or parallel polling would not be possible with such an api.

Reply:
it's just a grammatical remark: I'm pretty sure "in _the_ between" is more of a literary construct.